HW1/P4/P4_bfs.py

```diff
@@ -0,0 +1,85 @@
+import findspark
+findspark.init('/Users/Grace/spark-1.5.0-bin-hadoop2.6/')
+import pyspark
+import copy
+
+MAX_DIAMETER = 10
+
+
+def BFS_with_diameter(graph, root_node):
+    queue = [root_node]
+    visited_nodes = dict()
+
+    for distance in range(0, MAX_DIAMETER+1):
+        neighbor = []
+        for node in queue:
+            if node not in visited_nodes:
+                visited_nodes[node] = distance
+                neighbor = neighbor + graph.lookup(node)[0]
```

Comment:
I think you were not supposed to use lookup, since it is one of these very expensive functions...